packages/block-library/src/block/test/edit.native.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import { cleanup, fireEvent, waitFor } from '@testing-library/react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { getBlockTypes, unregisterBlockType } from '@wordpress/blocks';
+import { fetchRequest } from '@wordpress/react-native-bridge';
+
+/**
+ * Internal dependencies
+ */
+import { initializeEditor } from 'test/helpers';
+import { getReusableBlock } from './fixture';
+
+beforeAll( () => {
+	// Setup API fetch as it's done when initializing the editor
+	require( '@wordpress/react-native-editor/src/api-fetch-setup' ).default();
+	// Register all core blocks
+	require( '@wordpress/block-library' ).registerCoreBlocks();
+} );
+
+afterAll( () => {
+	// Clean up registered blocks
+	getBlockTypes().forEach( ( block ) => {
+		unregisterBlockType( block.name );
+	} );
+} );
+
+afterEach( cleanup );
+
+// The skipped tests below pass but are currently skipped because multiple
+// async findBy* queries currently cause errors in test output
+// https://git.io/JYYGE
+// eslint-disable-next-line jest/no-disabled-tests
+describe.skip( 'Reusable block', () => {
+	it( 'renders warning when the block does not exist', async () => {
+		// We have to use different ids because entities are cached in memory.
+		const id = 1;
+		const initialHtml = `<!-- wp:block {"ref":${ id }} /-->`;
+		const endpoint = `/wp/v2/blocks/${ id }`;
+
+		const result = await initializeEditor( { initialHtml } );
+		const { getBlockAtPosition, getByText } = result;
+
+		// Wait for the block to be fetched.
+		await waitFor( () => {
+			const isBlockFetched = fetchRequest.mock.calls.find( ( call ) =>
+				call[ 0 ].startsWith( endpoint )
+			);
+			expect( isBlockFetched ).toBeTruthy();
+		} );
+
+		expect( getBlockAtPosition( 'Reusable block' ) ).toBeDefined();
+		expect(
+			getByText( 'Block has been deleted or is unavailable.' )
+		).toBeDefined();
+
+		expect( global.getEditorHtml() ).toBe( initialHtml );
+	} );
+
+	it( 'renders block content', async () => {
+		// We have to use different ids because entities are cached in memory.
+		const id = 2;
+		const initialHtml = `<!-- wp:block {"ref":${ id }} /-->`;
+		const endpoint = `/wp/v2/blocks/${ id }`;
+
+		fetchRequest.mockImplementation( ( path ) => {
+			if ( path.startsWith( endpoint ) ) {
+				return Promise.resolve(
+					JSON.stringify( getReusableBlock( id ) )
+				);
+			}
+		} );
+
+		const result = await initializeEditor( { initialHtml } );
+		const { getBlockAtPosition } = result;
+
+		// Wait for the block to be fetched.
+		await waitFor( () => {
+			const hasFetchedBlock = fetchRequest.mock.calls.find( ( call ) =>
+				call[ 0 ].startsWith( endpoint )
+			);
+			expect( hasFetchedBlock ).toBeTruthy();
+		} );
+
+		const reusableBlock = await getBlockAtPosition( 'Reusable block' );
+		const innerBlockListWrapper = reusableBlock.findByProps( {
+			accessibilityLabel: 'block-list-wrapper',
+		} );
```

Comment:
The problem is that there're multiple elements with `block-list-wrapper` so, for getting the one inside the reusable block, instead of querying the result of the render, I'm querying the [ReactInstance](https://reactjs.org/docs/test-renderer.html#testinstance) returned from `getBlockAtPosition` call.

I'm wondering if there would be a better solution for the case when we want to query elements that are nested on other elements, wdyt?

Reply:
If I am understanding correctly, I believe we should be able to use [RNTL's `within` utility](https://callstack.github.io/react-native-testing-library/docs/api/#within-getqueriesforelement). Possibly something like the following. 

```js
const parentScope = within( getByA11yLabel( 'block-list-wrapper' ) );
const innerBlockListWrapper = parentScope.getByA11yLabel(
    'block-list-wrapper'
);
```

Reply:
Oh, I totally overlooked that utility 😄 , thanks for sharing it ❤️ ! I'll update this part with the code you suggested.

Reply:
`within` works like charm, I updated the tests in this [commit](https://github.com/WordPress/gutenberg/pull/32038/commits/e99d7cf2a7c577596171f77b6c658ea6dd2581d8) to use this function.